File: msgraph-mail/ruby/graphrubyv4/utilities/users/inference_classification/inference_classification_override.rb
require 'microsoft_kiota_abstractions'
require_relative '../entity'

module Graphrubyv4::Utilities::Users::InferenceClassification
    class InferenceClassificationOverride < Graphrubyv4::Utilities::Users::Entity
        include MicrosoftKiotaAbstractions::Parsable
        @classify_as
        @sender_email_address
        ## 
        ## Gets the classifyAs property value. 
        ## @return a inference_classification_type
        ## 
        def  classify_as
            return @classify_as
        end
        ## 
        ## Gets the senderEmailAddress property value. 
        ## @return a email_address
        ## 
        def  sender_email_address
            return @sender_email_address
        end
        ## 
        ## The deserialization information for the current model
        ## @return a i_dictionary
        ## 
        def get_field_deserializers() 
            return {
                "classify_as" => lambda {|o, n| o.classify_as = n.get_enum_value(inference_classification_type) },
                "sender_email_address" => lambda {|o, n| o.sender_email_address = n.get_object_value(email_address) },
            }
        end
        ## 
        ## Serializes information the current object
        ## @param writer Serialization writer to use to serialize this model
        ## @return a void
        ## 
        def serialize(writer) 
            super.serialize(writer)
            writer.write_enum_value("classify_as", self.classify_as)
            writer.write_object_value("sender_email_address", self.sender_email_address)
        end
        ## 
        ## Sets the classifyAs property value. 
        ## @param value Value to set for the classifyAs property.
        ## @return a void
        ## 
        def  classify_as=(classifyAs)
            @classify_as = (classifyAs)
        end
        ## 
        ## Sets the senderEmailAddress property value. 
        ## @param value Value to set for the senderEmailAddress property.
        ## @return a void
        ## 
        def  sender_email_address=(senderEmailAddress)
            @sender_email_address = (senderEmailAddress)
        end
    end
end

File: msgraph-mail/java/utilities/src/main/java/graphjavav4/utilities/models/microsoft/graph/InferenceClassificationOverride.java
package graphjavav4.utilities.models.microsoft.graph;

import com.microsoft.kiota.serialization.Parsable;
import com.microsoft.kiota.serialization.ParseNode;
import com.microsoft.kiota.serialization.SerializationWriter;
import java.util.function.BiConsumer;
import java.util.HashMap;
import java.util.Map;
import java.util.Objects;
public class InferenceClassificationOverride extends Entity implements Parsable {
    private InferenceClassificationType _classifyAs;
    private EmailAddress _senderEmailAddress;
    /**
     * Instantiates a new inferenceClassificationOverride and sets the default values.
     * @return a void
     */
    public InferenceClassificationOverride() {
        super();
    }
    /**
     * Gets the classifyAs property value. 
     * @return a inferenceClassificationType
     */
    @javax.annotation.Nullable
    public InferenceClassificationType getClassifyAs() {
        return this._classifyAs;
    }
    /**
     * Gets the senderEmailAddress property value. 
     * @return a emailAddress
     */
    @javax.annotation.Nullable
    public EmailAddress getSenderEmailAddress() {
        return this._senderEmailAddress;
    }
    /**
     * The deserialization information for the current model
     * @return a Map<String, BiConsumer<T, ParseNode>>
     */
    @javax.annotation.Nonnull
    public <T> Map<String, BiConsumer<T, ParseNode>> getFieldDeserializers() {
        return new HashMap<>(super.getFieldDeserializers()) {{
            this.put("classifyAs", (o, n) -> { ((InferenceClassificationOverride)o).setClassifyAs(n.getEnumValue(InferenceClassificationType.class)); });
            this.put("senderEmailAddress", (o, n) -> { ((InferenceClassificationOverride)o).setSenderEmailAddress(n.getObjectValue(EmailAddress.class)); });
        }};
    }
    /**
     * Serializes information the current object
     * @param writer Serialization writer to use to serialize this model
     * @return a void
     */
    public void serialize(@javax.annotation.Nonnull final SerializationWriter writer) {
        Objects.requireNonNull(writer);
        super.serialize(writer);
        writer.writeEnumValue("classifyAs", this.getClassifyAs());
        writer.writeObjectValue("senderEmailAddress", this.getSenderEmailAddress());
    }
    /**
     * Sets the classifyAs property value. 
     * @param value Value to set for the classifyAs property.
     * @return a void
     */
    public void setClassifyAs(@javax.annotation.Nullable final InferenceClassificationType value) {
        this._classifyAs = value;
    }
    /**
     * Sets the senderEmailAddress property value. 
     * @param value Value to set for the senderEmailAddress property.
     * @return a void
     */
    public void setSenderEmailAddress(@javax.annotation.Nullable final EmailAddress value) {
        this._senderEmailAddress = value;
    }
}
